lib/index.ts

```diff
@@ -0,0 +1,291 @@
+import fs = require('fs');
+import path = require('path-posix');
+import symlinkOrCopy = require('symlink-or-copy');
+import Logger = require('heimdalljs-logger');
+
+import Entry from './entry';
+import {
+  sortAndExpand,
+  validateSortedUnique
+} from './util';
+
+declare namespace FSTree {
+  export type Operand = 'unlink' | 'rmdir' | 'create' | 'change' | 'mkdir';
```

Comment:
Entry type is not reexported on FSTree namespace and has to be deeply imported.

Reply:
I'm an idiot and can't figure out how to export that :P

Reply:
I'd suggest importing from your other module like
```ts
import FSEntry from './entry';
```
and then in your namespace

```ts
declare namespace FSTree {
  // ...
  export class Entry extends FSEntry {}
  // ...
}
```
which will leave you with a `FSTree.Entry` that serves as both the interface for the instance, and the factory. You can use this throughout the rest of this module

```ts
new FSTree.Entry(...) // would fail if FSTree.Entry were just a type alias
const entries: FSTree.Entry[] // would still work if FSTree.Entry were just a type alias
```